elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.security;
+
+import com.yahoo.elide.core.filter.Operator;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.AndFilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.filter.expression.NotFilterExpression;
+import com.yahoo.elide.core.filter.expression.OrFilterExpression;
+import com.yahoo.elide.security.checks.InlineCheck;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Check for FilterExpression
+ *
+ * @param <T> Type of class
+ */
+public abstract class FilterExpressionCheck<T> extends InlineCheck<T> {
+
+    public abstract FilterExpression getFilterExpression(RequestScope requestScope);
+
+    @Override
+    public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
+        FilterExpression filterExpression = getFilterExpression(requestScope);
+        return applyFilterExpressionToObject(object, filterExpression);
+    }
+
+    private boolean applyFilterExpressionToObject(T object, FilterExpression filterExpression) {
+        if (filterExpression == null) {
+            return true;
+        }
+        if (AndFilterExpression.class.isAssignableFrom(filterExpression.getClass())) {
```

Comment:
Generally not advised to use isAssignableFrom and instanceof when it can be avoided.   A visitor would be better here.